### ISSUE TITLE
how to fix  "tf.nightly-gpu" caused "nan" problem

### ISSUE THREAD

**Issue by Q82822** [tensorflow/tensorflow#22213]
### System information
Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
OS Platform and Distribution (e.g., Linux Ubuntu 16.04):Linux Ubuntu 16.04
Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
TensorFlow installed from (source or binary): source
TensorFlow version (use command below): 1.11.0 (use tf-night-gpu)
Python version:2.7
Bazel version (if compiling from source):
GCC/Compiler version (if compiling from source):
CUDA/cuDNN version: 9.0 /7.1
GPU model and memory:
Exact command to reproduce:

### Describe the problem
I used `tf-night-gpu` in my project code, but it will be made "nan" at the next batch train.
it only the  `tf-night-gpu` will happen, it is can work in tensorflow1.10.

### Source code / logs
the minimal code at here , it is just part of my project code. the input is word_index.
the batch size = 64
use tf-nightly-gpu.
platform = jupyter notebook.
in first batch size was worked, but in next batch train will be "nan". but this problem only happen in "tf-nightly-gpu"
```

class  RNN_Decoder(tf.keras.Model):
    def __init__(self, embedding_dim, units, vocab_size):
        super(RNN_Decoder, self).__init__()
        self.units = units
        self.embedding = tf.keras.layers.Embedding(vocab_size, embedding_dim)
    def call(self, x, features, hidden):
        print('embedding_input:  ', x)
        x = self.embedding(x)
        print('embedding_output:  ', x)

decoder = RNN_Decoder(embedding_dim, units, vocab_size)

for epoch in range(20):
    hidden = decoder.reset_state(batch_size=64)
    dec_input = tf.expand_dims([tokenizer.word_index['<start>']] * BATCH_SIZE, 1)
        with tf.GradientTape() as tape:
            features = encoder(img_tensor)
            for i in range(1, target.shape[1])
            predictions, hidden, _ = decoder(dec_input, features, hidden)
```

---- first batch train input and output------
```
('embedding_input ', <tf.Tensor: id=4586198, shape=(64, 1), dtype=int32, numpy=
array([[3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3]], dtype=int32)>)
('Embedding_output:  ', <tf.Tensor: id=4586379, shape=(64, 1, 256), dtype=float32, numpy=
array([[[ 0.01097491, -0.02041122,  0.04823625, ..., -0.0155656 ,
          0.01521399, -0.0496642 ]],

       [[ 0.01097491, -0.02041122,  0.04823625, ..., -0.0155656 ,
          0.01521399, -0.0496642 ]],

       [[ 0.01097491, -0.02041122,  0.04823625, ..., -0.0155656 ,
          0.01521399, -0.0496642 ]],

       ...,

       [[ 0.01097491, -0.02041122,  0.04823625, ..., -0.0155656 ,
          0.01521399, -0.0496642 ]],

       [[ 0.01097491, -0.02041122,  0.04823625, ..., -0.0155656 ,
          0.01521399, -0.0496642 ]],

       [[ 0.01097491, -0.02041122,  0.04823625, ..., -0.0155656 ,
          0.01521399, -0.0496642 ]]], dtype=float32)>)
```

the next batch train , have same keras.layers.Embedding_input:

```
`Epoch 1 Batch 0 Loss 2.0415
(''keras.layers.Embedding_input:: ', <tf.Tensor: id=4607289, shape=(64, 1), dtype=int32, numpy=
array([[3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3],
       [3]], dtype=int32)>)
(''keras.layers.Embedding_output::  ', <tf.Tensor: id=4607374, shape=(64, 1, 256), dtype=float32, numpy=
array([[[nan, nan, nan, ..., nan, nan, nan]],

       [[nan, nan, nan, ..., nan, nan, nan]],

       [[nan, nan, nan, ..., nan, nan, nan]],


       [[nan, nan, nan, ..., nan, nan, nan]],

       [[nan, nan, nan, ..., nan, nan, nan]],

       [[nan, nan, nan, ..., nan, nan, nan]]], dtype=float32)>)`
```

**Comment by karmel** [comment]
Sorry, but I am not able to run or replicate with the code snippet above; can you produce a minimal example that demonstrates the issue?

**Comment by Q82822** [comment]
I'm copy from [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/eager/python/examples/generative_examples/image_captioning_with_attention.ipynb) ypu just use tf-nightly-gpu running this code. 
and output it  `embedding`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @karmel: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @karmel: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by MarkDaoust** [comment]
It looks like the notebook was relying on undefined behavior for Out of Vocabulary values, and this undefined behavior changed in r 1.11.

I'll see what I can find for a fix.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi @Q82822 ! 1.x versions are not supported any more. You can use our [migration document](https://www.tensorflow.org/guide/migrate) for using  1.x codes in 2.8 version. Thanks!
